Adding GitHub Actions for CI, runs the build process, partially resolves #68

aryan-99 opened the pull request:
Added `actions.yml ` under `.github/workflows` that checks out code, fixes yarn issue using `tabularasa.sh`, builds and starts container, partially resolves #68

aryan-99 committed:
Adding potential CI

Added yml file for CI workflow
Untested

aryan-99 committed:
Testing Actions with Docker Compose

- Changed file name
- Changed file contents to build docker container using compose

aryan-99 committed:
Updated actions.yml for yarn integrity issue

actions.yml now runs tabularasa.sh before building

aryan-99 committed:
Fixed path for executing tabularasa

aryan-99 committed:
Fixed permission issue for tabularasa

aryan-99 committed:
Potential fix for permissions issue

aryan-99 committed:
FIx for tabularasa permission for actions.yml

aryan-99 committed:
Adding step to give tabularasa.sh executable permission

aryan-99 committed:
Changes in tabularasa.sh and actions.yml

- Commented out docker build and run commands from tabularasa.sh
- Added docker-compose build and up as separate steps in actions.yml

aryan-99 committed:
Added -d flag to up command in actions.yml

The -d flag allows the container to start in a detached container and finish the build for actions

kbuffardi commented on .github/workflows/actions.yml:
```diff
@@ -0,0 +1,19 @@
+        uses: actions/checkout@v2
+      - name: Permissions for tabularasa.sh
+        run: chmod +x ./tabularasa.sh
+      - name: Workaround for yarn integrity issue
```
Did you try the workflow without this step? The script is only meant for us to use when we really need to start from scratch (Tabula Rasa = "blank slate") and ideally, we don't have to do this every time we start a new CI action.

We should remove this (and the previous step to set permissions) if it isn't absolutely necessary.

kbuffardi commented on .github/workflows/actions.yml:
```diff
@@ -0,0 +1,19 @@
+        run: chmod +x ./tabularasa.sh
+      - name: Workaround for yarn integrity issue
+        run: ./tabularasa.sh
+      - name: Docker Compose Build
```
I believe `docker-compose up` will build if there is not a container already built, so we may be able to remove this step.

aryan-99 committed:
Potential fix for yarn integrity issue without tabularasa.sh

- Removed permissions change and running tabularasa.sh for actions.yml
- Commented out a repeat statement that sets check_yarn_integrity to true in webpacker.yml

aryan-99 committed:
Changing event to push on primary branch for actions.yml

aryan-99 committed:
Undo comments in tabularasa.sh

aryan-99 commented:
Closing pull request, changes are covered in a new iteration of the `github-actions` branch. Will open a new pull request.
